tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -86,12 +86,37 @@ class RFieldBase {
    friend struct ROOT::Experimental::Internal::RFieldCallbackInjector; // used for unit tests
    using ReadCallback_t = std::function<void(void *)>;
 
+protected:
+   /// A functor to release the memory acquired by GenerateValue (memory and constructor).
+   /// This implementation works for types with a trivial destructor. More complex fields implement a derived deleter.
+   /// The deleter is operational without the field object and thus can be used to destruct/release a value after
+   /// the field has been destructed.
+   class RDeleter {
```

Comment:
What is the main motivation to keep the `RDeleter` and derived classes nested in the various `RField`s? As opposed to having a separate header file?

Reply:
I'd say locality, the destruction code is analogous to the `RField::GenerateValue` code and therefore I thought it makes sense to keep them close to each other.